test/arrays/methods/vectorMethods.chpl

```diff
@@ -0,0 +1,99 @@
+/* Tests for domain-modifying methods commonly referred to as vector methods */
+
+// push_back
+{
+  var A: [1..0] int;
+  A.push_back(0);
+  A.push_back([1,2,3,4]);
+
+  assert(A.equals([0,1,2,3,4]));
+}
+
+// push_front
+{
+  var A: [1..0] int;
+
+  A.push_front(4);
+  A.push_front([0,1,2,3]);
+
+  assert(A.equals([0,1,2,3,4]));
+}
+
+// insert
+{
+  var A: [1..0] int;
+
+  A.insert(1, 4);
+  A.insert(1, [0,3]);
+  A.insert(2, [1,2]);
+
+  assert(A.equals([0,1,2,3,4]));
+}
+
+//
+// Type coercion
+//
+
+// push_back
+{
+  var A: [1..0] real;
+  A.push_back(0);
+  A.push_back([1,2,3,4]);
+
+  assert(A.equals([0,1,2,3,4]));
+}
+
+// push_front
+{
+  var A: [1..0] real;
+
+  A.push_front(4);
+  A.push_front([0,1,2,3]);
+
+  assert(A.equals([0,1,2,3,4]));
+}
+
+// insert
+{
+  var A: [1..0] real;
+
+  A.insert(1, 4);
+  A.insert(1, [0,3]);
+  A.insert(2, [1,2]);
+
+  assert(A.equals([0,1,2,3,4]));
+}
+
+//
+// Arrays of arrays
+//
+
+// push_back
+{
+  var A: [1..0] [1..2] int;
+  const B: [1..3] [1..2] int = [[1,1],[2,2],[3,3]];
+
+  A.push_back(B);
+
+  assert(A.equals(B));
+}
+
+// push_front
+{
+  var A: [1..0] [1..2] int;
+  const B: [1..3] [1..2] int = [[1,1],[2,2],[3,3]];
+
+  A.push_front(B);
+
+  assert(A.equals(B));
+}
+
+// insert
+{
+  var A: [1..0] [1..2] int;
+  const B: [1..3] [1..2] int = [[1,1],[2,2],[3,3]];
+
+  A.insert(1, B);
+
+  assert(A.equals(B));
+}
```

Comment:
What if A does a push_back/push_front/insert with itself? Or a slice of itself?

Reply:
It crashes and burns. *The plot thickens.*

Reply:
I think that should be undefined, so crash and burn is fine.

Reply:
@daviditen  - How strongly do you feel about this being undefined? I have this working now in my upcoming commit..

Reply:
If it's working, parallel, and doesn't use a temp array then I'm good with that.

Reply:
Got `push_*` to work, but we'll leave `insert()` for a rainy day.